### PR TITLE
Document a set of Kafka integration recipes

### PR DESCRIPTION
**Pull request by cescoffier**
* Kafka with Hibernate
* Kafka with Hibernate Reactive
* Kafka to SSE
* HTTP to Kafka

### CODE REVIEW

File: docs/src/main/asciidoc/kafka.adoc [hunk 1]
```diff
@@ -1548,6 +1548,368 @@ kafka.bootstrap.servers=...
...
+<3> Send the message to Kafka, the `send` method returns a `CompletionStage`
+
+The endpoint sends the passed payload (from a `POST` HTTP request) to the emitter.
+The emitter's channel is mapped to a Kafka topic in the `application.properties` file:
+
+[source, properties]
+----
+mp.messaging.outgoing.kafka.connector=smallrye-kafka
+mp.messaging.outgoing.kafka.topic=my-topic
+mp.messaging.outgoing.kafka.value.serializer=org.apache.kafka.common.serialization.StringSerializer
```
Comment (Ladicek) [inline docs/src/main/asciidoc/kafka.adoc:1594]: Not required, is autodetected from the `@Channel` injection point :-)

File: docs/src/main/asciidoc/kafka.adoc [hunk 2]
```diff
@@ -1548,6 +1548,368 @@ kafka.bootstrap.servers=...
...
+        return emitter.send(Record.of("my-key", payload));    // <2>
+    }
+}
+----
+<1> Note the usage of an `Emitter<Record<K, V>>`
+<2> Create the record using `Record.of(k, v)`
+
+=== Persisting Kafka messages with Hibernate with Panache
+
+To persist objects, received from Kafka into a database, you can use Hibernate with Panache.
```
Comment (Ladicek) [inline docs/src/main/asciidoc/kafka.adoc:1640]:
```suggestion
To persist objects received from Kafka into a database, you can use Hibernate with Panache.
```

File: docs/src/main/asciidoc/kafka.adoc [hunk 3]
```diff
@@ -1548,6 +1548,368 @@ kafka.bootstrap.servers=...
...
+[source, properties]
+----
+mp.messaging.incoming.fruits.connector=smallrye-kafka
+mp.messaging.incoming.fruits.value.deserializer=org.acme.FruitDeserializer
+----
+
+Check <<jackson-serialization>> for more detail about the usage of Jackson with Kafka.
+You can also use Avro.
+
+=== Writing entities managed by Hibernate to Kafka
```
Comment (Ladicek) [inline docs/src/main/asciidoc/kafka.adoc:1806]: Do we want a Hibernate Reactive variant of this section?

File: docs/src/main/asciidoc/kafka.adoc (hunk 1 above)
Comment (Ladicek) [reply docs/src/main/asciidoc/kafka.adoc:1594]: This would also apply to all other examples, but let's not complicate this.

**Review by Ladicek — APPROVED** [review]

**Review by gsmet — CHANGES_REQUESTED** [review]
Very nice work, once the small thing noted by @Ladicek is fixed, we can merge it.

**Review by ozangunalp — APPROVED** [review]
Great addition to the Kafka documentation

File: docs/src/main/asciidoc/kafka.adoc (hunk 3 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/kafka.adoc:1805]: Added.

File: docs/src/main/asciidoc/kafka.adoc (hunk 2 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/kafka.adoc]: Fixed.

**Review by gsmet — APPROVED** [review]
Great, thanks!

**Pull request merged**